### PR TITLE
Add `overlay` MaterialStateProperty property to `Slider`

### LINKED ISSUE #112864: Add missing `overlay` property and support Material states for `SliderTheme.overlay`
While working on https://github.com/flutter/flutter/issues/111451.

I came across some issues where:

1. `Slider` doesn't have an `overlay` property. Currently, `Slider` uses `activeColor` for the track, thumb, and overlay.
We should be able to use `activeColor` for track and thumb and dedicated `overlay` property based on `MaterialStateProperty` for overlay itself (hovered, focused, dragged states).

2. `SliderTheme.overlay` doesn't support Material states. While it's not a `MaterialStatePropety` it should support material states.

Adding this code does nothing in the `Slider`
```dart
        sliderTheme: SliderThemeData(
          overlayColor: MaterialStateColor.resolveWith(
            (Set<MaterialState> states) {
              if (states.contains(MaterialState.hovered)) {
                return Colors.blue;
              }
              if (states.contains(MaterialState.focused)) {
                return Colors.green;
              }
              if (states.contains(MaterialState.dragged)) {
                return Colors.red;
              }
              return const Color(0x00000000);
            },
          ),
        ),
```

https://user-images.githubusercontent.com/48603081/193858731-3f927be9-beb1-4135-ad8a-6e92c011d34a.mov

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/112864 

This PR does the following:

1. Add `overlay` MaterialStateProperty property.
2. Add support for material states to `SliderTheme.overlay` and new `Slider.overlay` property (move `states` above so `sliderTheme.copyWith` can access the states).
3. Move the `Slider uses the right theme colors for the right components` theme test from `slider_test.dart` to `slider_theme_test.dart`
4. Remove some blank lines.

This is preliminary work for https://github.com/flutter/flutter/issues/111451
 
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/material/slider_test.dart [hunk 1]
```diff
@@ -626,252 +625,6 @@ void main() {
     log.clear();
   });
 
-  testWidgets('Slider uses the right theme colors for the right components', (WidgetTester tester) async {
```
Comment (HansMuller) [inline packages/flutter/test/material/slider_test.dart:629]: I'm assuming these tests have just been moved?

File: packages/flutter/lib/src/material/slider.dart [hunk 2]
```diff
@@ -734,6 +740,16 @@ class _SliderState extends State<Slider> with TickerProviderStateMixin {
     const SliderComponentShape defaultValueIndicatorShape = RectangularSliderValueIndicatorShape();
     const ShowValueIndicator defaultShowValueIndicator = ShowValueIndicator.onlyForDiscrete;
 
+    final Set<MaterialState> states = <MaterialState>{
+      if (!_enabled) MaterialState.disabled,
+      if (_hovering) MaterialState.hovered,
+      if (_focused) MaterialState.focused,
+      if (_dragging) MaterialState.dragged,
+    };
+    final MouseCursor effectiveMouseCursor = MaterialStateProperty.resolveAs<MouseCursor?>(widget.mouseCursor, states)
```
[the remark this answers is not in the document]
Comment (TahaTesser) [reply packages/flutter/lib/src/material/slider.dart]: `effectiveMouseCursor` doesn't need to be moved, it was simply moved together with `states`, and restored it.

File: packages/flutter/lib/src/material/slider.dart [hunk 3]
```diff
@@ -413,6 +415,10 @@ class Slider extends StatefulWidget {
   /// (like the native default iOS slider).
   final Color? thumbColor;
 
+  /// The highlight color that's typically used to indicate that
+  /// the slider is focused, hovered, or dragged.
+  final MaterialStateProperty<Color?>? overlayColor;
```
[the remark this answers is not in the document]
Comment (TahaTesser) [reply packages/flutter/lib/src/material/slider.dart:425]: Added explanation :)

File: packages/flutter/test/material/slider_test.dart (hunk 1 above)
Comment (TahaTesser) [reply packages/flutter/test/material/slider_test.dart:629]: This is a theme test residing `slider_test.dart`, it should be under `slider_theme_test.dart`.
 I'll need to update this test for M3 and M2 groups in a new future PR under `slider_theme_test.dart`

Comment (HansMuller) [reply packages/flutter/test/material/slider_test.dart:629]: OK

**Comment by HansMuller** [comment]
Lots of test failures that didn't appear to be related to this PR. I've tried restarting them. You may need to resync with master

**Review by HansMuller — COMMENTED** [review]
LGTM

**Review by HansMuller — APPROVED** [review]

**Pull request merged**